dotCMS/src/main/java/com/dotcms/security/secret/ServiceIntegrationAPIImpl.java

```diff
@@ -0,0 +1,554 @@
+import com.dotmarketing.business.CacheLocator;
+import com.dotmarketing.business.LayoutAPI;
+import com.dotmarketing.business.UserAPI;
+import com.dotmarketing.exception.DotDataException;
+import com.dotmarketing.exception.DotSecurityException;
+import com.dotmarketing.portlets.contentlet.business.HostAPI;
+import com.dotmarketing.util.Config;
+import com.dotmarketing.util.Logger;
+import com.dotmarketing.util.UtilMethods;
+import com.fasterxml.jackson.annotation.JsonAutoDetect.Visibility;
+import com.fasterxml.jackson.annotation.PropertyAccessor;
+import com.fasterxml.jackson.databind.ObjectMapper;
+import com.fasterxml.jackson.dataformat.yaml.YAMLFactory;
+import com.google.common.annotations.VisibleForTesting;
+import com.google.common.collect.ImmutableList;
+import com.liferay.portal.model.User;
+import com.liferay.util.StringPool;
+import io.vavr.Tuple2;
+import java.io.File;
+import java.io.IOException;
+import java.io.InputStream;
+import java.nio.file.DirectoryStream;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.nio.file.Paths;
+import java.util.Collections;
+import java.util.HashSet;
+import java.util.List;
+import java.util.Map;
+import java.util.Map.Entry;
+import java.util.Optional;
+import java.util.Set;
+import java.util.function.Function;
+import java.util.function.Supplier;
+import java.util.stream.Collectors;
+
+public class ServiceIntegrationAPIImpl implements ServiceIntegrationAPI {
+
+    static final String INTEGRATIONS_PORTLET_ID = "integration-services";
+    private static final String HOST_SECRET_KEY_SEPARATOR = ":";
+    private static final String DOT_GLOBAL_SERVICE = "dotCMSGlobalService";
+    private static final String SERVICE_INTEGRATION_DIR_PATH_KEY = "SERVICE_INTEGRATION_DIR_PATH_KEY";
+
+    private UserAPI userAPI;
+    private LayoutAPI layoutAPI;
+    private HostAPI hostAPI;
+    private SecretsStore secretsStore;
+
+
+    private final ObjectMapper jsonMapper = new ObjectMapper()
+            //.enable(SerializationFeature.INDENT_OUTPUT)
+            .setVisibility(PropertyAccessor.FIELD, Visibility.ANY);
+
+    private final ObjectMapper ymlMapper = new ObjectMapper(new YAMLFactory())
+            //.enable(SerializationFeature.INDENT_OUTPUT)
+            .setVisibility(PropertyAccessor.FIELD, Visibility.ANY)
+            .findAndRegisterModules();
+
+    private String toJsonString(final ServiceSecrets object) throws DotDataException {
+        try {
+            return jsonMapper.writeValueAsString(object);
+        } catch (IOException e) {
+            throw new DotDataException(e);
+        }
+    }
+
+    private ServiceSecrets readJson(final String json) throws DotDataException {
+        try {
+            return jsonMapper.readValue(json, ServiceSecrets.class);
+        } catch (IOException e) {
+            throw new DotDataException(e);
+        }
+    }
+
+    private ServiceSecrets readJson(final char[] chars) throws DotDataException {
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Consider using varargs for methods or constructors which take an array the last parameter.](https://app.codacy.com/manual/dotCMS/core/pullRequest?prid=4623512)